src/machine/flash.go

```diff
@@ -89,38 +93,50 @@ func (fl *FlashBuffer) Read(p []byte) (n int, err error) {
 	return
 }
 
-// Write data to a FlashBuffer.
+// Write data to a FlashBuffer. You must pass in a slice containing a multiple
+// of the WriteBlockSize() for the target device, or the data written will be
+// padded to fill in any missing space.
 func (fl *FlashBuffer) Write(p []byte) (n int, err error) {
-	// any new pages needed?
+	// any new blocks needed?
 	// NOTE probably will not work as expected if you try to write over page boundary
 	// of pages with different sizes.
-	pagesize := uintptr(fl.b.EraseBlockSize())
+	blocksize := uintptr(fl.b.EraseBlockSize())
 
 	// calculate currentPageBlock relative to fl.start, meaning that
 	// block 0 -> fl.start
-	// block 1 -> fl.start + pagesize
-	// block 2 -> fl.start + pagesize*2
+	// block 1 -> fl.start + blocksize
+	// block 2 -> fl.start + blocksize*2
 	// ...
-	currentPageBlock := (fl.start + fl.offset - FlashDataStart()) + (pagesize-1)/pagesize
-	lastPageBlockNeeded := (fl.start + fl.offset + uintptr(len(p)) - FlashDataStart()) + (pagesize-1)/pagesize
+	currentPageBlock := (fl.offset + (blocksize - 1)) / blocksize
+	lastPageBlockNeeded := (fl.offset + uintptr(len(p)) + (blocksize - 1)) / blocksize
```

Comment:
Why add (blocksize-1) here?
If fl.offset is 0, the correct value 0 is returned; if fl.offset is 1, 1 is returned. But isn't it correct to return 0 even when fl.offset is 1?